Feature: add hero header component

rpollock03 opened the pull request:
Resolves #57 - adds new hero header component, using MUI components and responsive design. Sign in component to be added separately. 

![Screenshot 2022-10-28 at 09 25 31](https://user-images.githubusercontent.com/49661708/198541962-19bf4f1e-e2e8-4b81-81bd-d24f4421f783.png)
![Screenshot 2022-10-28 at 09 25 53](https://user-images.githubusercontent.com/49661708/198541974-5ec15ef1-7f3a-4d17-84aa-42f88bde7073.png)

piotrostr commented on components/HeroHeader1.tsx:
```diff
@@ -0,0 +1,115 @@
+
+const Star = styled(StarIcon)`
+  color: #fec84b;
+`
```
nit: add newline for correct spacing

piotrostr commented on components/HeroHeader1.tsx:
```diff
@@ -0,0 +1,115 @@
+  font-family: 'Plus Jakarta Sans'
+  line-height: 24px;
+  margin-left: 4px;
+`
```
same here

piotrostr commented on components/HeroHeader1.tsx:
```diff
@@ -0,0 +1,115 @@
+import StarIcon from "@mui/icons-material/Star"
+import Image from "next/image"
+
+const Heading = styled(Typography)`
```
Rather than creating a one-off styled Typography component, it would be more elegant to add global styling and use Typography variants.

See https://mui.com/material-ui/customization/typography/#variants

piotrostr replied:
Be sure not to break any of the existing Typography'ies though! :)

rpollock03 replied:
Good idea, I have updated to create custom global styling using typography variants (added to the same object that you created to add the Jakarta font)

piotrostr commented:
And one last thing, please be sure not to close the pull requests, rather rebase or perform a merge commit.

See https://www.atlassian.com/git/tutorials/merging-vs-rebasing

rpollock03 commented:
@piotrostr grateful for your review, I have addressed your comments. In particular, and as you suggested, I have replaced one-off styled components with global styling using Typography variants, which I have included in the same place you added the Jakarta font (heroHeading and heroSubheading).

piotrostr commented on components/HeroHeader1.tsx:
```diff
@@ -0,0 +1,108 @@
+const AverageReview = styled(Typography)`
+  color: #3a6942;
+  font-size: 16px;
+  font-weight: 600;
```
Could we also move this to `theme.ts`? The naming can just be different levels of h, like h1, h2, ..., h6 to be reused throughout the app.

rpollock03 replied:
this is quite a specific use case (green font etc) so not sure it makes sense to use h6 etc here as we won't be using elsewhere much, but I have moved to themes.ts as you have suggested.

rpollock03 commented:
> Perfect, one more thing - please don't include numbers in file names nor the component names.
> 
> `HeroHeader1` -> `HeroHeader`

Done.

piotrostr reviewed (changes requested):
Looking good! Now the absolute last thing to do would be to check if the typography like one on the homepage appears elsewhere in the page design and then, adjust it to be the right heading/paragraph/span.

There might be a part of the design with all of the possible colors and fonts as well as the logos and so on somewhere on Figma, it could be useful to check for the above there as well.

Lastly, having checked that, please modify the default Typography variants rather than create the custom variants. 

For example (no exactly but along those lines): 
`reviewAverage` -> `span` but say with `color: gray`
or
`heroHeading` -> `h2`

Let me know if that makes sense.
